core/src/Borrower.sol

```diff
@@ -116,8 +116,11 @@ contract Borrower is IUniswapV3MintCallback {
 
     receive() external payable {}
 
+    /// @dev This allows re-initialization to a different `owner`
     function initialize(address owner) external {
-        require(slot0.owner == address(0));
+        uint256 ownerAndState = _loadSlot0() & 0xff0000000000000000000000ffffffffffffffffffffffffffffffffffffffff;
```

Comment:
Masks out the `unleashLiquidationTime`. Lowest 160 bits are the `owner`, and highest 8 are the `state` flag.

We want `state == State.Ready`, which is 0. So we should be left with just the address, and can assert that it's either 0 or msg.sender.

Reply:
A bit confused. According to this [article](https://medium.com/@mweiss.eth/solidity-and-evm-bit-shifting-and-masking-in-assembly-yul-942f4b4ebb6a), `f = 1111`. 

```
struct Slot0 {
    address owner;
    uint88 unleashLiquidationTime;
    State state;
}
```
Shouldn't the mask be `0xffffffffffffffffffffffffffffffffffffffff0000000000000000000000ff`?

Reply:
See the following snippet from the article; in our case `state` is like variable `F`, so it should be first in our mask.

<img width="713" alt="image" src="https://github.com/aloelabs/aloe-ii/assets/17186559/93fe3df7-d0b9-41a3-b4e1-83b9de9e57a2">